diff --git a/index.d.ts b/index.d.ts
index 549a2852da41aa369c746564622ce3bfbd2a3339..b8bed15b0e1debeb984d47602840299b51f07b32 100644
--- a/index.d.ts
+++ b/index.d.ts
@@ -26,6 +26,12 @@ export interface Actions {
     r9kbetaoff(channel: string): Promise<[string]>;
     raw(message: string): Promise<[string]>;
     say(channel: string, message: string): Promise<[string]>;
+    /**
+     * Sends a reply to a message.
+     * @param parentMessageId The ID of the message to reply to.
+     * @param message The message to send.
+     */
+    reply(parentMessageId: string, message: string): Promise<void>;
     slow(channel: string, length?: number): Promise<[string]>;
     slowoff(channel: string): Promise<[string]>;
     subscribers(channel: string): Promise<[string]>;
@@ -139,11 +145,14 @@ export interface Events {
 }
 
 export interface ClientBase {
+    channels: string[];
+    opts: Options;
     getChannels(): string[];
     getOptions(): Options;
     getUsername(): string;
     isMod(channel: string, username: string): boolean;
     readyState(): "CONNECTING" | "OPEN" | "CLOSING" | "CLOSED";
+    isConnected(): boolean;
     on(event: any, listener: any): Client;
     addListener(event: any, listener: any): Client;
     removeListener(event: any, listener: any): Client;
